modin/experimental/core/execution/native/implementations/omnisci_on_native/exchange/dataframe_protocol/column.py

```diff
@@ -0,0 +1,488 @@
+            pandas_dtype_to_arrow_c(dtype),
+            dtype.byteorder,
+        )
+
+    @property
+    def describe_categorical(self) -> Dict[str, Any]:
+        dtype = self._pandas_dtype
+
+        if dtype != "category":
+            raise RuntimeError(
+                f"Column 'dtype' has to be categorical to be able to dectribe categiries, met: {dtype}"
+            )
+
+        ordered = dtype.ordered
+
+        # Category codes may change during materialization flow, so trigger
+        # materialization before returning the codes
+        self._materialize_actual_buffers()
+
+        # Although we can retrieve codes from pandas dtype, they're unsynced with
+        # the actual PyArrow data most of the time. So getting the mapping directly
+        # from the materialized PyArrow table.
+        col = self._pyarrow_table.column(0)
+        if len(col.chunks) > 1:
+            if not self._col._allow_copy:
+                raise RuntimeError("Copy required but 'allow_copy=False'")
+            col = col.combine_chunks()
+
+        col = col.chunks[0]
+        mapping = {index: value for index, value in enumerate(col.dictionary.tolist())}
+
+        return {
+            "is_ordered": ordered,
+            "is_dictionary": True,
+            "mapping": mapping,
+        }
+
+    @property
+    def describe_null(self) -> Tuple[ColumnNullType, Any]:
+        null_buffer = self._pyarrow_table.column(0).chunks[0].buffers()[0]
+        if null_buffer is None:
+            return (ColumnNullType.NON_NULLABLE, None)
+        else:
+            return (ColumnNullType.USE_BITMASK, 0)
+
+    @property
+    def null_count(self) -> int:
+        ncount = self._pyarrow_table.column(0).null_count
+        return ncount if ncount >= 0 else None
+
+    @property
+    def metadata(self) -> Dict[str, Any]:
+        return self._col.metadata
+
+    @property
+    def _pandas_dtype(self) -> np.dtype:
+        """
+        Get column's dtype representation in Modin DataFrame.
+
+        Returns
+        -------
+        numpy.dtype
+        """
+        return self._col._df.dtypes.iloc[0]
+
+    @property
+    def _arrow_dtype(self) -> pa.DataType:
+        """
+        Get column's dtype representation in underlying PyArrow table.
+
+        Returns
+        -------
+        pyarrow.DataType
+        """
+        return self._pyarrow_table.column(0).type
+
+    @property
+    def _pyarrow_table(self) -> pa.Table:
+        """
+        Get PyArrow table representing the column.
+
+        Returns
+        -------
+        pyarrow.Table
+        """
+        return self._col._pyarrow_table
+
+    def num_chunks(self) -> int:
+        return self._col.num_chunks()
+
+    def get_chunks(self, n_chunks: Optional[int] = None) -> Iterable["Column"]:
+        for chunk in self._col.get_chunks(n_chunks):
+            yield OmnisciProtocolColumn(chunk)
+
+    def get_buffers(self) -> Dict[str, Any]:
+        if self.num_chunks() != 1:
```

Comment:
Why?

Reply:
If the table is physically chunked there's no continuous buffer to return, in this case, we have to combine chunks into a single contiguous buffer, this logic is not yet implemented/tested.

Reply:
Doesn't arrow have an API to convert pyarrow table to pyarrow buffer? Do you plan to introduce that as part of a separate PR?

Reply:
PyArrow only can convert an Array instance into a buffer, each column is a ChunkedArray that consists of multiple Arrays (if chunked). There's a special method for PyArrow table that combines chunks into a single contiguous buffer ([`.combine_chunks()`](https://arrow.apache.org/docs/python/generated/pyarrow.Table.html#pyarrow.Table.combine_chunks)).

I've added the logic of chunks combining in the last commit. Now it's allowed to access column's buffers even if it's physically chunked.

Reply:
This doesn't look right. If a column has 3 chunks, for example, then when calling get_chunks we get 3 chunks. At the same time, when calling get_buffers we get 1 contiguous buffer, while buffers should 3. What do you think on this?

Reply:
Let's follow both of the scenarious:
1. Column is physically chunked and we access each chunk's buffer separately:
```python
>>> col.num_chunks()
3
>>> len(col._pyarrow_table.column(0).chunks)
3
>>> ch1, ch2, ch3 = col.get_chunks()
>>> ch1.num_chunks(), ch2.num_chunks(), ch3.num_chunks()
(1, 1, 1)
>>> ch1.get_buffers()["data"][0].ptr, ch2.get_buffers()["data"][0].ptr, ch3.get_buffers()["data"][0].ptr
(140593764278336, 140593764278720, 140593764279168) # Each chunk has its own buffer, so there's 3 different pointers
```
2. Column is physically chunked and we access the chunked column buffer. As there're 3 separate buffers, we have to copy the data of 3 chunks into a single contiguous buffer
```python
>>> col.num_chunks()
3
>>> len(col._pyarrow_table.column(0).chunks)
3
>>> col.get_buffers() # Combines chunks to provide single contiguous buffer
{'data': (Buffer({'bufsize': 104, 'ptr': 139876896911360, 'device': 'CPU'}), (<DTypeKind.INT: 0>, 64, 'l', '=')), 'validity': None, 'offsets': None}
>>> col.num_chunks()
1
>>> len(col._pyarrow_table.column(0).chunks)
1
```

Why the state is mutated during the second flow? It's to not to redo the combining every time we access `.get_buffers()` or some other method that require to materialize actual buffers (`.describe_categories`). Redoing combining every time may result into a strang behaviour:
```python
>>> ptr1 = col.get_buffers()["data"][0].ptr # Do chunks combining for the first time
>>> ptr2 = col.get_buffers()["data"][0].ptr # Do chunks combining again
>>> # Although we called exactly the same method, the results are different
>>> ptr1 == ptr2
>>> False
>>> mapping1 = col.describe_categories["mapping"] # Materialization is required, do chunks combining for the first time 
>>> mapping2 = col.describe_categories["mapping"] # Do chunks combining again
>>> # mapping1 != mapping2 in case categoricals are unordered
>>> mapping1 == mapping2
>>> False
```

Reply:
Let's leave as is.